src/cli/commands/db/init/init.spec.ts

```diff
@@ -0,0 +1,130 @@
+  DATA_API_BUILDER_DEFAULT_CONFIG_FILE_NAME,
+  DATA_API_BUILDER_DEFAULT_FOLDER,
+  // DATA_API_BUILDER_DEFAULT_REST_PATH,
+  // DATA_API_BUILDER_DEFAULT_SCHEMA_FILE_NAME,
+  // DEFAULT_DATA_API_BUILDER_SCHEMA_CONTENT,
+} from "../../../../core/constants";
+import fs from "fs";
+
+// Replace the imported functions with mocks for testing purposes
+jest.mock("../../../../core/dataApiBuilder", () => ({
+  getDataApiBuilderBinaryPath: jest.fn(() => "dataApiBuilderPath"),
+}));
+jest.mock("../../../../core", () => ({
+  fs: {
+    existsSync: jest.fn(() => false),
+    mkdirSync: jest.fn(),
+    writeFileSync: jest.fn(),
+  },
+  logger: {
+    log: jest.fn(),
+    error: jest.fn(),
+    warn: jest.fn(),
+    info: jest.fn(),
+  },
+  execFileCommand: jest.fn(),
+}));
+
+describe("init", () => {
+  beforeEach(() => {
+    // Reset the mocked functions before each test
+    jest.clearAllMocks();
+  });
+
+  it("should log an error and return if databaseType is undefined", async () => {
+    const options = {};
+
+    await init(options);
+
+    expect(logger.error).toHaveBeenCalledWith(
+      `--database-type is a required field. Please provide the type of the database you want to connect (mssql, postgresql, cosmosdb_nosql, mysql, cosmosdb_postgresql).`,
+      true
+    );
+  });
+
+  it("should log an error and return if databaseType is invalid", async () => {
+    const options = { databaseType: "invalidDatabaseType" };
+
+    await init(options);
+
+    expect(logger.error).toHaveBeenCalledWith(
+      `--database-type is a required field. Please provide the type of the database you want to connect (mssql, postgresql, cosmosdb_nosql, mysql, cosmosdb_postgresql).`,
+      true
+    );
+  });
+
+  it("should create the folder if it doesn't exist", async () => {
+    const options = { databaseType: "mssql" };
+    const fsExistsSyncSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(false);
+    const fsMkdirSyncSpy = jest.spyOn(fs, "mkdirSync").mockReturnValue("");
+
+    await init(options);
+
+    expect(fsExistsSyncSpy).toHaveBeenCalledTimes(2);
+    expect(fsMkdirSyncSpy).toHaveBeenCalledWith(expect.any(String));
+    expect(logger.log).toHaveBeenCalledWith(`Creating database connections configuration folder ${DATA_API_BUILDER_DEFAULT_FOLDER}`, "swa");
+  });
+
+  it("should not create the folder if it already exists", async () => {
+    const options = { databaseType: "mssql", folderName: "existingFolder" };
+    const fsExistsSyncSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+    const fsMkdirSyncSpy = jest.spyOn(fs, "mkdirSync").mockReturnValue("");
+
+    await init(options);
+
+    expect(fsExistsSyncSpy).toHaveBeenCalledTimes(2);
+    expect(fsMkdirSyncSpy).not.toHaveBeenCalled();
+    expect(logger.log).toHaveBeenCalledWith(`Folder existingFolder already exists, using that folder for creating data-api files`, "swa");
+  });
+
+  it("should log an error and return if the config file already exists", async () => {
```

Comment:
Shall we also add tests for CosmosDbnoSqlContainer/CosmosDbnoSqlDatabase/ related warn/error loggings ?

Reply:
added now